elasticapm/metrics/base_metrics.py

```diff
@@ -255,6 +294,40 @@ def val(self):
     def val(self, value):
         self._val = value
 
+    def reset(self):
+        self._val = 0
+
+
+class Timer(object):
+    __slots__ = ("name", "_val", "_count", "_lock", "reset_on_collect")
+
+    def __init__(self, name=None, reset_on_collect=False):
+        self.name = name
+        self._val = AtomicNumber()
+        self._count = AtomicNumber()
+        self._lock = threading.Lock()
+        self.reset_on_collect = reset_on_collect
+
+    def update(self, duration, count=1):
+        with self._lock:
+            self._val.inc(duration)
+            self._count.inc(count)
+
+    def reset(self):
+        with self._lock:
+            self._val.value = 0
+            self._count.value = 0
+
+    @property
+    def val(self):
+        with self._lock:
+            return self._val.value, self._count.value
+
+    @val.setter
+    def val(self, value):
```

Comment:
Maybe drop this method? I'm not sure there's a valid use case for it at the moment.